### PR TITLE
fix(web): prevent blank module numbers saving zero

### LINKED ISSUE #1173: fix(web): prevent blank Beast module numeric fields from saving zero
## Description
The Beast creation wizard's module numeric inputs can silently save `0` when an operator clears a field, even though the UI declares positive minimums. React number inputs temporarily produce an empty string while editing; the current handlers pass `Number(e.target.value)`, and `Number('')` is `0`. That lets blank Max DAG Depth, Parallel Task Limit, Max Iterations, or Reflection Interval values enter wizard state as zero, bypassing the visible `min` constraints until a later backend/schema failure or invalid runtime config.

## Affected locations
- `packages/franken-web/src/components/beasts/steps/step-modules.tsx:173-181` — Max DAG Depth uses `min={1}` but writes `Number(e.target.value)`, so clearing the field stores `0`.
- `packages/franken-web/src/components/beasts/steps/step-modules.tsx:185-193` — Parallel Task Limit has the same blank-to-zero behavior despite `min={1}`.
- `packages/franken-web/src/components/beasts/steps/step-modules.tsx:203-211` — Critique Max Iterations has the same blank-to-zero behavior despite `min={1}`.
- `packages/franken-web/src/components/beasts/steps/step-modules.tsx:264-272` — Heartbeat Reflection Interval has the same blank-to-zero behavior despite `min={10}`.

## Reproduction / reasoning path
1. Open the Create Agent wizard and enable Planner, Critique, or Heartbeat module configuration.
2. Focus one of the numeric fields and clear the current value while editing.
3. The browser emits `event.target.value === ''`.
4. The handler calls `Number('')`, which stores `0` in the wizard step state.
5. The visible field no longer reflects the documented positive minimum, and downstream launch/review code can receive an invalid zero-valued module config.

## User impact
Operators can accidentally create invalid or surprising Beast module configs by simply clearing a number field during normal editing. The bug is especially risky for runtime controls like planner depth, parallelism, critique iterations, and heartbeat interval because a zero value may disable expected behavior, cause backend validation failures late in the flow, or produce confusing launch errors disconnected from the field that caused them.

## Recommendation
Parse numeric inputs with explicit blank handling and bounds validation. Keep blank fields as `undefined`/draft text until valid, clamp or reject out-of-range numbers before saving wizard state, and surface inline validation errors before launch/review. Add regression tests that clear each numeric field and assert the wizard does not persist `0` for fields whose minimum is positive.

## Acceptance criteria
- Clearing any module numeric field no longer stores `0` when the field minimum is greater than zero.
- The wizard shows an inline validation error or restores a safe default before the operator can launch/review invalid module config.
- Tests cover blank input for Planner Max DAG Depth, Planner Parallel Task Limit, Critique Max Iterations, and Heartbeat Reflection Interval.
- The final agent create payload cannot contain zero for these positive-only module settings unless the schema explicitly allows it.

Duplicate check: read `/home/pfkagent/.hermes/state/frankenbeast_open_issues_cache.json` and searched cache/live GitHub for `StepModules`, `maxDagDepth`, `parallelTaskLimit`, `reflectionInterval`, `Number(e.target.value)`, `empty numeric`, and `maxIterations`. Existing #1012 covers broader Create Agent validation before invalid runtime configs; this ticket is narrowed to the concrete blank-to-zero conversion in module numeric fields and the missing field-level regression coverage.

### PR DESCRIPTION
**Pull request by djm204**
## Summary
- Prevent blank Beast module numeric fields from being coerced through `Number('')` into `0`.
- Remove cleared positive numeric module config values so the wizard falls back to safe defaults.
- Clamp out-of-range values on blur and sanitize launch config as a final guard against zero/blank/non-finite positive-only module numeric payloads.

## Test Plan
- `TMPDIR=/home/pfkagent/dev/resolve-wt/issue-1173/.tmp npm test -- tests/components/beasts/steps/step-modules.test.tsx tests/components/beasts/wizard-launch-config.test.ts`
- `TMPDIR=/home/pfkagent/dev/resolve-wt/issue-1173/.tmp npm run typecheck` after building `@franken/types`
- `TMPDIR=/home/pfkagent/dev/resolve-wt/issue-1173/.tmp npm run build`

Closes #1173

### CODE REVIEW

**Comment by djm204** [comment]
@codex review

File: packages/franken-web/src/components/beasts/steps/step-modules.tsx [hunk 1]
```diff
@@ -268,7 +281,7 @@ function renderModuleConfig(
               min={10}
               max={600}
               value={(config.reflectionInterval as number) ?? 60}
-              onChange={(e) => update('reflectionInterval', Number(e.target.value))}
+              onChange={(e) => update('reflectionInterval', parseBoundedNumberInput(e.target.value, 10, 600))}
```
[the remark this answers is not in the document]
Comment (djm204) [reply packages/franken-web/src/components/beasts/steps/step-modules.tsx]: Fixed in 47d61e2e by moving bounds clamping from `onChange` to `onBlur`, so transient edits like `3` → `30` are preserved while the field is active. I also added launch-config sanitization as a final guard so positive-only module numeric settings cannot reach the launch payload as zero/blank/non-finite values.

**Comment by djm204** [comment]
@codex review
